openml/runs/functions.py

```diff
@@ -1094,6 +1094,19 @@ def list_runs(offset=None, size=None, id=None, task=None, setup=None,
         List of found runs.
     """
 
+    if id is not None and (not type(id) is list):
+        raise Exception('id must be of type list.')
+        return None
```

Comment:
Returning None after raising an exception is not necessary. 
As soon as the exception is raised, control is returned to the calling function.